### DISCUSSION TITLE
Question about installation

### DISCUSSION THREAD

**Discussion by opierpaoli** [raphaelquast/EOmaps#73] [Q&A]
Hi, 

I am trying to install and run some code using EOmaps, and while I was successfully able to do this on one computer, I am not able to replicate the same figures with a copy of the same conda environment on my other laptop. When I use the plot_map function I get this error: RuntimeWarning: invalid value encountered in subtract delta = stop - start which makes the figure extents out of bounds. I am using python 3.8.13, numpy 1.21.5, cartopy 0.20.2, and EOmaps 3.5. Any ideas on how to fix this or is there a .yml file that could be provided for me with the correct versions of the required packages? 

Thanks!

**Comment by raphaelquast** [comment]
Hi, 
sorry to hear that but from what you report here it's quite hard to know where your error is coming from...

- what do you mean by a "copy of the same conda environment"... did you export a .yml file and installed a new env from that?
- can you provide the code or at least the full error traceback message?

----

In general, the fastest way to get EOmaps running is the following:

- install `mamba` to speed up the process
  `conda install -c conda-forge mamba`
- create a new env with eomaps (and matplotlib to get all the backends)
  `mamba create -n NAME -c conda-forge eomaps matplotlib`
  (`NAME` is your env name and the python-version can be anything above 3.7)

**Reply by opierpaoli** [reply]
Hi! 
To answer: what do you mean by a "copy of the same conda environment"... did you export a .yml file and installed a new env from that?
Yes, I tried this, but it didn't work. But I've also used the same commands on 2 different computers and one environment
 works and the other has a numpy core error.

Here is the full error message:
\path\to\env\site-packages\numpy\core\function_base.py:134: RuntimeWarning: invalid value encountered in subtract delta = stop - start

this error appears once I use the plot_map function. I can still draw coastlines and other features.

I tried installing using :
>conda create -n NAME -c conda-forge eomaps matplotlib

and I still get the same issue.

**Reply by opierpaoli** [reply]
just tried with mamba as well, and I'm still getting the same error. Specifically when using plot_map.

**Comment by raphaelquast** [comment]
hey, ok so that seems like a quite strange problem...

it certainly originates from some call to `np.linspace`:
https://github.com/numpy/numpy/blob/0eaa40db3a2f6bef26c0d7cc28eacf4f6d461917/numpy/core/function_base.py#L134

However, since I don't have any clue to what's happening yet, I can only give a few more suggestions on how we could proceed to find out what's going on:


- concerning the error-message... this is not an error, this is a warning... in theory it should not break the code...
  The only way I can get the same warning is with something like: `np.array([np.inf, 1, 2,3]) - np.array([np.inf, 1,2,3])`
  (e.g. only undefined forms such as `np.inf - np.inf` give such a warning)
  From this I have some basic suggestions that might solve this 
  (however none of them explain why it happens only on one computer and not on the other):
  - are you sure your input-data does not contain points at infinity (e.g. `np.inf`) ?
  - are you sure you provide the correct `crs` for the coordinates?
    (`m.set_data(data, xcoord, ycoord, crs=???)`  otherwise reprojections might result in `np.inf`)
  

- can you provide a minimal reproducible example that I could try on my side?
- what operating system are you working on? (is it the same on both computers?)
- did you try with a fresh conda? (maybe the issue is originating from the conda-installation...)
  - you don't need to uninstall your existing setup... just use [miniconda](https://docs.conda.io/en/latest/miniconda.html)  
    to install another conda-instance and then try this again:
    ```
    conda install -c conda-forge mamba
    mamba create -n Name -c conda-forge eomaps matpltolib
    ```

**Reply by opierpaoli** [reply]
Hi, so I believe that the error has to do with the operating system. I realized earlier today that this was not the only time that code has worked on one laptop and not the other, it happened when I was setting the extent of images using Basemap as well. In particular,  there seems to be issues with infinite or nan values. The operating systems are only 2 builds away from each other one is windows 10 19044 and the other is 19042 (this one doesn't work). I've tried everything and can't seem to get it to work properly. If you have any thoughts why this is happening I'd love to know!

Thanks for your help!

**Reply by raphaelquast** [reply]
While this is of course possible, I've never heard of a windows-OS version causing such a problem.... 
After all the conda-installation should be very much self-contained and the only OS-dependency that 
I can think of are the precompiled C-extensions of libraries such as `numpy`...
(which might be resolved with a completely fresh miniconda env as suggested above)

Unfortunately I'm also on 19044 so I can't reproduce your errors on my side...


Can you provide some minimal set of code that causes this error ? 
For example does this already result in a problem:

```python
from eomaps import Maps
import numpy as np
lon, lat = np.meshgrid(*np.linspace([-50,-30], [50, 30], 100).T)

m = Maps()
m.set_data(lon, lon, lat)
m.plot_map()
```

.... and just out of curiosity (since you mentioned setting the plot-extent)... can you also try with:
```python
m.plot_map(set_extent=False)
```

**Reply by opierpaoli** [reply]
I was using a fresh install of Anaconda, but I could try miniconda as well. I am also able to use the code you just sent and everything seems to be working properly. 

You were actually helping me out with code over on StackOverflow with this post: [https://stackoverflow.com/questions/71804525/cartopy-coastlines-not-lining-up-with-imshow-projection/71956219#71956219](url)

```
from eomaps import Maps
import numpy as np
from pyhdf.SD import SD, SDC

data = SD("Copy of greenland_PM01_L2.2007061414150564.058140.1.46-public-release.hdf", SDC.READ)

#get the actual coordinates of your datapoints
lon, lat = data.select("subset_lon")[:], data.select("subset_lat")[:]
#mask nan-values
mask = np.logical_and(np.isfinite(lon), np.isfinite(lat))
lon, lat = lon[mask], lat[mask]

#get the colors you want to use
R = data.select("mband07")[:][mask]
G = data.select("mband02")[:][mask]
B = data.select("mband01")[:][mask]
RGB = list(zip(R, G, B)) # a list of rgb-tuples

#plot the data
m = Maps(Maps.CRS.Sinusoidal())
m.add_feature.preset.coastline()
m.set_shape.rectangles(radius=3100, radius_crs=Maps.CRS.Sinusoidal())
# use "dummy" values since we provide explicit colors
m.set_data(data=np.empty(lon.shape), xcoord=lon, ycoord=lat, crs=4326)
m.plot_map(set_extent=True, color=RGB)
```

I've since changed the code slightly to use a different projection. There's code that someone else sent using Basemap and I have the same issues where the set_extent works on one computer and not the other. With the computer that does work, the figures I am able to make with this library are amazing, so thank you again!

**Reply by raphaelquast** [reply]
aaah, OK I see,  nice to hear that (at least on one PC) the library works nicely 😅 
There is in fact a part in the code I've provided that might be OS-dependent.... `np.empty(lon.shape)` !
(it just initializes an empty array without caring about what's been allocated at the memory-location before...)
maybe try and replace it with something that is for sure not a very large number... such as `np.full(lon.shape, 0)`

**Reply by raphaelquast** [reply]
In the next release of EOmaps (v3.6 coming soon...) "coordinate-only" datasets can be plotted as well (e.g. using something like `m.set_data(None, lon, lat)`) but in v3.5 that's not yet possible

**Reply by opierpaoli** [reply]
I changed it and I get the same warning as before. I guess it has to be something with how the different os's behave with infinite numbers then?
